lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1210 @@
+"""
+Process DETs for the UW Reopening (Husky Coronavirus Testing) REDCap projects.
+"""
+import re
+import click
+import json
+import logging
+from uuid import uuid4
+from typing import Any, Callable, Dict, List, Mapping, Match, NamedTuple, Optional, Union, Tuple
+from cachetools import TTLCache
+from dateutil.relativedelta import relativedelta
+from enum import Enum
+from id3c.db.session import DatabaseSession
+from id3c.cli.command.etl import redcap_det
+from id3c.cli.command.geocode import get_response_from_cache_or_geocoding
+from id3c.cli.command.location import location_lookup
+from id3c.cli.command.de_identify import generate_hash
+from id3c.cli.redcap import is_complete, Record as REDCapRecord
+from seattleflu.id3c.cli.command import age_ceiling
+from .redcap import normalize_net_id
+from .redcap_map import map_sex, map_symptom, UnknownVaccineResponseError
+from .fhir import *
+from . import race, first_record_instance, required_instruments
+
+
+LOG = logging.getLogger(__name__)
+
+
+class HuskyProject():
+    id: int
+    lang: str
+    api_token_env_var: str
+    command_name: str
+
+    def __init__(self, project_id: int, lang: str, api_token_env_var: str, command_name: str) -> None:
+        self.id = project_id
+        self.lang = lang
+        self.api_token_env_var = api_token_env_var
+        self.command_name = command_name
+
+PROJECTS = [
+        HuskyProject(23854, "en", "REDCAP_API_TOKEN", "uw-reopening")
+    ]
+
+LANGUAGE_CODE = {
+project.id: project.lang
+    for project in PROJECTS }
+
+class CollectionMethod(Enum):
+    SWAB_AND_SEND = 'swab_and_send'
+    KIOSK = 'kiosk'
+
+class EventType(Enum):
+    ENROLLMENT = 'enrollment'
+    ENCOUNTER = 'encounter'
+
+REVISION = 0
+
+REDCAP_URL = 'https://redcap.iths.org/'
+INTERNAL_SYSTEM = "https://seattleflu.org"
+ENROLLMENT_EVENT_NAME = "enrollment_arm_1"
+ENCOUNTER_EVENT_NAME = "encounter_arm_1"
+
+REQUIRED_ENROLLMENT_INSTRUMENTS = [
+    'eligibility_screening',
+    'consent_form',
+    'enrollment_questionnaire'
+]
+
+
+def command_for_each_project(function):
+    """
+    A decorator to register one redcap-det subcommand per REDCap project, each
+    calling the same base *function*.
+
+    Used for side-effects only; the original *function* is unmodified.
+    """
+    for project in PROJECTS:
+        help_message = "Process REDCap DETs for Husky Coronavirus Testing (UW reopening)"
+
+        redcap_det.command_for_project(
+            name = project.command_name,
+            redcap_url = REDCAP_URL,
+            project_id = project.id,
+            raw_coded_values = True,
+            revision = REVISION,
+            help = help_message,
+            include_incomplete = True)(function)
+
+    return function
```

Comment:
I did this intentionally because I suspect that we'll end up having more than one REDCap project for the UW Reopening. We may never have more than one project, I get that.